### PR TITLE
RWA-1906 improve doc for search endpoint

### PR DESCRIPTION
**Pull request by mkayad**
### JIRA link (if applicable) ###

[RWA-1906](https://tools.hmcts.net/jira/browse/RWA-1906)

### Change description ###
see [RWA-1906](https://tools.hmcts.net/jira/browse/RWA-1906)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java [hunk 1]
```diff
@@ -8,9 +8,10 @@ public enum SearchParameterKey {
     USER("user"),
     JURISDICTION("jurisdiction"),
     STATE("state"),
-    TASK_ID("taskId"),
-    TASK_TYPE("taskType"),
-    CASE_ID("caseId"),
+    TASK_TYPE_CAMEL_CASE("taskType"),
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java:11]: Is anyone using this one?  If not then we should only support the `snake_case`

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java [hunk 2]
```diff
@@ -1299,8 +1299,8 @@ public void should_return_a_200_with_multiple_tasks_in_search_results_when_avail
 
         //Claim a task to set state to assigned
         given.iClaimATaskWithIdAndAuthorization(taskId3,
-                                                caseworkerCredentials.getHeaders(),
-                                                HttpStatus.NO_CONTENT);
+            caseworkerCredentials.getHeaders(),
+            HttpStatus.NO_CONTENT);
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java:1303]: Are these formatting changes required?

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java [hunk 3]
```diff
@@ -121,7 +121,7 @@ public void should_return_a_200_with_search_results_and_correct_properties() {
             .body("tasks.warnings", everyItem(equalTo(false)))
             .body("tasks.case_management_category", everyItem(equalTo("Protection")))
             .body("tasks.work_type_id", everyItem(equalTo("hearing_work")))
-            .body("tasks.permissions.values", everyItem(equalToObject(List.of("Read", "Own", "CompleteOwn","CancelOwn","Claim"))))
+            .body("tasks.permissions.values", everyItem(equalToObject(List.of("Read", "Own", "CompleteOwn", "CancelOwn", "Claim"))))
```
Comment (raghera) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java:124]: more formatting

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java [hunk 4]
```diff
@@ -8,9 +8,10 @@ public enum SearchParameterKey {
...
+    TASK_TYPE("task_type"),
+    CASE_ID_CAMEL_CASE("caseId"),
+    CASE_ID("case_id"),
     //R2 should be snake_case only,
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java:15]: Maybe we should update the comment at the top of the file to say something like:
`If adding a new field please ensure they are added as snake_case only and not camelCase. The camelCase format is deprecated and will be removed.`

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceSpecificationTest.java [hunk 5]
```diff
@@ -226,7 +226,7 @@ void should_build_task_query_with_all_parameters() {
             new SearchParameterList(JURISDICTION, SearchOperator.IN, singletonList("IA")),
             new SearchParameterList(STATE, SearchOperator.IN, singletonList("ASSIGNED")),
             new SearchParameterList(LOCATION, SearchOperator.IN, singletonList("location")),
-            new SearchParameterList(CASE_ID, SearchOperator.IN, singletonList("caseId")),
+            new SearchParameterList(CASE_ID, SearchOperator.IN, singletonList("case_id")),
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceSpecificationTest.java:229]: Probably want to ensure there's a test that covers that the camelCase still works.

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java (hunk 2 above)
Comment (mkayad) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerCFTTest.java:1303]: no looks like I have a different checkstyle. will fix it

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java (hunk 3 above)
Comment (mkayad) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskSearchControllerTest.java:124]: will remove

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java (hunk 4 above)
Comment (mkayad) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java:15]: will do

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java (hunk 1 above)
Comment (mkayad) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchParameterKey.java:11]: good point, ExUI does not use, will check our functional tests and remove the camelcase

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceSpecificationTest.java (hunk 5 above)
Comment (mkayad) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceSpecificationTest.java:230]: done

**Review by raghera — DISMISSED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**